services/language_translation/v1.js

```diff
@@ -13,34 +13,169 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  **/
-
-module.exports = function(RED) {
+module.exports = function (RED) {
   var watson = require('watson-developer-cloud');
   var cfenv = require('cfenv');
+  var fs = require('fs');
+  var fileType = require('file-type');
+  var request = require('request');
+  var temp = require('temp');
+  temp.track();
 
-  var services = cfenv.getAppEnv().services,
-    service;
+  var services = cfenv.getAppEnv().services;
 
-  var username, password;
+  var username, password, language_translation;
 
-  var service = cfenv.getAppEnv().getServiceCreds(/language translation/i)
+  var service = cfenv.getAppEnv().getServiceCreds(/language translation/i);
 
   if (service) {
     username = service.username;
     password = service.password;
   }
- 
-  RED.httpAdmin.get('/watson-translate/vcap', function(req, res) {
-    res.json(service ? {bound_service: true} : null);
+
+  RED.httpAdmin.get('/watson-translate/vcap', function (req, res) {
+    res.json(service ? {
+      bound_service: true
+    } : null);
+  });
+
+  RED.httpAdmin.get('/watson-translate/models', function (req, res) {
+    language_translation = watson.language_translation({
+      username: username,
+      password: password,
+      version: 'v2'
+    });
+    language_translation.getModels({}, function (err, models) {
+      if (err)
+        res.json(err);
+      else
+        res.json(models);
+    });
+  });
+
+  RED.httpAdmin.get('/watson-translate/languages', function (req, res) {
+    language_translation = watson.language_translation({
+      username: username,
+      password: password,
+      version: 'v2'
+    });
+    language_translation.getIdentifiableLanguages(null,
+      function (err, languages) {
+        if (err)
+          res.json(err);
+        else
+          res.json(languages);
+      });
   });
 
   function SMTNode(config) {
     RED.nodes.createNode(this, config);
     var node = this;
 
-    this.on('input', function(msg) {
+    this.doTranslate = function (msg, model_id) {
+      language_translation = watson.language_translation({
+        username: username,
+        password: password,
+        version: 'v2'
+      });
+
+      node.status({
+        fill: "blue",
+        shape: "dot",
+        text: "requesting"
+      });
+      language_translation.translate({
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected indentation of 8 space characters but found 10.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2396889184/issues/source?bid=3225115&fileBranchId=3231190#l88)